### PR TITLE
Raise WorkspaceChanged with a new SourceText even if we have a new snapshot without changes

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Merging this into another branch so we can deliver it into dev15.7.x-vs-deps.

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
@heejaechang That other branch didn't have CI enabled. Looks like this is failing:

```
System.ArgumentException: changes
   at Microsoft.CodeAnalysis.Text.TextChangeEventArgs..ctor(SourceText oldText, SourceText newText, IEnumerable`1 changes) in /_/src/Compilers/Core/Portable/Text/TextChangeEventArgs.cs:line 26
   at Microsoft.CodeAnalysis.Text.Extensions.TextBufferContainer.OnTextContentChanged(Object sender, TextContentChangedEventArgs args) in /_/src/EditorFeatures/Text/Extensions.TextBufferContainer.cs:line 117
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.RaiseEvent[TArgs](Object sender, EventHandler`1 eventHandlers, TArgs args)
```

Looks like we assumed changes is always empty. Not sure if it makes sense to just weaken that, or say there's a change of zero-length so it makes that happy.

File: src/Compilers/Core/Portable/Text/TextChangeEventArgs.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ public class TextChangeEventArgs : EventArgs
         /// <param name="changes">A non-empty set of ranges for the change.</param>
```
Comment (jasonmalinowski) [inline src/Compilers/Core/Portable/Text/TextChangeEventArgs.cs]: Docs would need changing. Any idea if anything in the compiler relied on this?

Comment (heejaechang) [reply src/Compilers/Core/Portable/Text/TextChangeEventArgs.cs]: so looked around. so it is only used only for
http://source.roslyn.io/#Microsoft.CodeAnalysis/Text/SourceTextContainer.cs,21

and in compiler, text never changes, so there is no one consuming the event.

**Comment by jasonmalinowski** [comment]
Well, I approve of this change but since I originated the PR I can't actually approve it. So, uh....

**Comment by jasonmalinowski** [comment]
@dotnet-bot retest windows_debug_unit64_prtest please.

**Comment by jasonmalinowski** [comment]
@dotnet-bot retest windows_debug_unit64_prtest please.

**Comment by heejaechang** [comment]
@dotnet-bot retest windows_debug_unit64_prtest please.

**Comment by heejaechang** [comment]
I am not sure why build is getting timed out

**Pull request merged**